daemon/src/main.rs

```diff
@@ -192,3 +200,12 @@ fn ask(question: &str) -> Result<bool> {
         bail!("Failed to read answer");
     }
 }
+
+fn has_git_remote(path: &Path) -> bool {
+    if let Ok(repo) = git2::Repository::open(path) {
```

Comment:
I realized that this heuristic doesn't work if someone wants to pair program on a subdirectory of a git repo. `Repository::open` will not "traverse up", like Git itself does, but I think that's good. But sometimes you might have a huge monorepo and only want to pair-program on a sub-tree. In our case, we could also easily "just" pair on `daemon` or so sometimes.

The heuristics will assume it should persist in this case.

Reply:
Hmm... my feeling is that in the sub-tree-pairing case I would also like Ethersync to not persist, so that the document histories are "ephemeral".

I once had an unusual setup where my home directory *was* a Git directory (for dotfiles), so there that might be a bit weird.

But maybe I'd slightly prefer switching to `discover()`? So we'd check "is our directory part of a Git repo with a remote"?